pkg/ottl/ottlfuncs/func_contains.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ContainsArguments[K any] struct {
+	Target ottl.PSliceGetter[K]
+	Item   ottl.Getter[K]
+}
+
+func NewContainsFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Contains", &ContainsArguments[K]{}, createContainsFunction[K])
+}
+
+func createContainsFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ContainsArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ContainsFactory args must be of type *ContainsArguments[K]")
+	}
+
+	return contains(args.Target, args.Item), nil
+}
+
+func contains[K any](target ottl.PSliceGetter[K], itemGetter ottl.Getter[K]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		slice, sliceErr := target.Get(ctx, tCtx)
+		if sliceErr != nil {
+			return nil, sliceErr
+		}
+		item, itemErr := itemGetter.Get(ctx, tCtx)
+		if itemErr != nil {
+			return nil, itemErr
+		}
+
+		for i := 0; i < slice.Len(); i++ {
+			val := slice.At(i).AsRaw()
+			if val == item {
+				return true, nil
+			}
+		}
```

Comment:
Although functionally correct and generic enough to cover multiple data types, this approach can become prohibitively expensive for performance sensitive applications under load. For example, if this approach was used for supporting routing-connector in open-telemetry collector to inspect 100+ string values while processing 100,000 spans per second kind of workload. A map would be much more efficient for such kind of conditions.

Reply:
just to clarify, when i say multiple data types.... I meant to appreciate that it can handle integer, strings, booleans alike....  However I am concerned that outer type has to be a slice only... which may not be the optimal choice for heavy workloads of say, simple string based look up.

Reply:
I understand your concern, but I'm not sure how to do it with a map. If we want to handle Contains that accept a slice as input, I don't see how to convert it to a map in linear time.  

If we accepted `map` as input in Contains, it would be easy to get an efficient search, but I'm not sure how to do it with slice.